vassal-app/src/main/java/VASSAL/configure/ConfigureTree.java

```diff
@@ -2401,40 +2721,93 @@ private boolean checkSearchTarget(SearchTarget st, String searchString) {
+      if (showName) {
         final String objectName = c.getConfigureName();
-        if (objectName != null && checkString(objectName, searchString)) {
+        if (objectName != null && checkString(objectName, regexPattern)) {
           return true;
         }
       }
 
-      if (searchParameters.isMatchTypes() || !searchParameters.isMatchAdvanced()) {
+      if (showTypes) {
         final String className = getConfigureName(c.getClass());
-        if ((className != null) && checkString(className, searchString)) {
+        if ((className != null) && checkString(className, regexPattern)) {
           return true;
         }
       }
 
-      if ((searchParameters.isMatchNames() && searchParameters.isMatchTypes()) || !searchParameters.isMatchAdvanced()) {
+      // Selecting names includes description in detection.
+      if (showName) {
         if (c instanceof ComponentDescription) {
           final String desc = ((ComponentDescription) c).getDescription();
-          if ((desc != null) && checkString(desc, searchString)) {
+          if ((desc != null) && checkString(desc, regexPattern)) {
             return true;
           }
         }
       }
 
-      if (!searchParameters.isMatchAdvanced()) {
+      if (searchParameters.isMatchSimple()) {
         return false;
       }
 
-      // From here down we are only searching inside of SearchTarget objects (Piece/Prototypes, or searchable AbstractConfigurables)
+      // For some reason, a pdf or text component is a SearchTarget (with no returns) even though overall Help Menu is not,
+      // so we test for Help Menu first, guarding against possibility of null node...
+      try {
+        //  Special processing to include select items in full search despite not being a SearchTarget
+        if (getConfigureName((Configurable) node.getParent()).equals(getString("Editor.Documentation.component_type"))) {
+          // [Help Menu]
+          // Menu is also the Name, so only catch here if not already caught as Name
+          if (!showName && (searchParameters.isMatchFull() || searchParameters.isMatchMenus())) {
+            return checkString(c.getConfigureName(), regexPattern);
+          }
+          if (searchParameters.isMatchFull() || searchParameters.isMatchExpressions()) {
+            // content will be categorised as Expressions... each is a separate component
+            final String startPage = c.getAttributeValueString("startingPage");
+            if (startPage != null) return checkString(startPage, regexPattern);
+
+            final String pdfFile = c.getAttributeValueString("pdfFile");
+            if (pdfFile != null) return checkString(pdfFile, regexPattern);
+
+            final String textFile = c.getAttributeValueString("textFile");
+            if (textFile != null) return checkString(textFile, regexPattern);
+          }
+        }
+      }
+      catch (Exception NullPointerException) {
```

Comment:
* Fix the NPE, don't catch it.
* Narrow Exception to whatever is actually thrown.